src/invite/greeter.rs

```diff
@@ -0,0 +1,362 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 (eventually AGPL-3.0) 2016-present Scille SAS
+
+use pyo3::prelude::*;
+
+use crate::{
+    backend_connection::AuthenticatedCmds,
+    data::SASCode,
+    enumerate::UserProfile,
+    ids::{DeviceLabel, HumanHandle, InvitationToken},
+    invite::error::InviteExc,
+    local_device::LocalDevice,
+    runtime::FutureIntoCoroutine,
+};
+
+#[pyclass]
+pub(crate) struct UserGreetInitialCtx(pub Option<libparsec::core::UserGreetInitialCtx>);
+
+crate::binding_utils::gen_proto!(UserGreetInitialCtx, __repr__);
+
+#[pymethods]
+impl UserGreetInitialCtx {
+    #[new]
+    fn new(cmds: &AuthenticatedCmds, token: InvitationToken) -> Self {
+        Self(Some(libparsec::core::UserGreetInitialCtx::new(
+            cmds.0.as_ref().clone(),
+            token.0,
+        )))
+    }
+
+    fn do_wait_peer(&mut self) -> FutureIntoCoroutine {
+        let ctx = self
+            .0
+            .take()
+            .expect("UserGreetInitialCtx has been consumed");
```

Comment:
Maybe we should raise an exception on the python code instead ?

Reply:
Is it important ?

Reply:
there should be a comment near `struct UserGreetInitialCtx` definition explaining why an option is needed, and in which case it can become `None`

We cannot just yolo panic if we encounter a `None` here ! We need to know precisely in which case we will end up here and why this is okay

Reply:
Done

Reply:
Raise RuntimeError instead

Reply:
Done